src/service/viewport-impl.js

```diff
@@ -302,6 +304,29 @@ export class Viewport {
   }
 
   /**
+   * Scrolls element into view much like Element. scrollIntoView does but
+   * in the AMP/Viewer environment. Adds animation for the sccrollIntoView
+   * transition.
+   *
+   * @param {!Element} element
+   * @param {number=} duration
+   * @param {string=} curve
+   * @return {!../../animation.AnimationPlayer}
```

Comment:
As mentioned above: you only need to return `AnimationPlayer` if you are planning to allow canceling this animation. This is pretty unusual. To convert animation player to a promise, simply call `.then()` on it w/o arguments.